TimesSquare/TSQCalendarDayButton.m

```diff
@@ -8,31 +8,223 @@
 
 #import "TSQCalendarDayButton.h"
 
-@implementation TSQCalendarDayButton
+@interface TSQCalendarDayButton ()
 
+@property (nonatomic, strong) UILabel *subtitleLabel;
+@property (nonatomic, strong) UILabel *subtitleSymbolLabel;
 
-/*
-// Only override drawRect: if you perform custom drawing.
-// An empty implementation adversely affects performance during animation.
-- (void)drawRect:(CGRect)rect {
-    // Drawing code
-}
-*/
+@property (nonatomic, strong) UIImageView *iconImageView;
+
+@end
+
+@implementation TSQCalendarDayButton
+
+static const CGFloat TSQCalendarRowCellMaxSubtitleHeight = 18.0f;
+static const CGFloat TSQCalendarRowCellSubtitleBuffer = 15.0f;
 
 - (instancetype)initWithFrame:(CGRect)frame
 {
     self = [super initWithFrame:frame];
     if (self) {
+        // default button type to normal day
+        self.type = CalendarButtonTypeNormalDay;
+
         [self setTitleEdgeInsets:UIEdgeInsetsMake(-10, 0, 0, 0)];
-        self.subtitleLabel = [[UILabel alloc] initWithFrame:CGRectMake(10, 50, 65, 18)];
+        self.subtitleLabel = [[UILabel alloc] initWithFrame:CGRectZero];
         self.subtitleLabel.textAlignment = NSTextAlignmentCenter;
         self.subtitleLabel.userInteractionEnabled = NO;
+        self.subtitleLabel.adjustsFontSizeToFitWidth = NO;
+        self.subtitleLabel.lineBreakMode = NSLineBreakByTruncatingTail;
         [self addSubview:self.subtitleLabel];
         
-        self.subtitleSymbolLabel = [[UILabel alloc] initWithFrame:CGRectMake(75, 50, 8, 18)];
+        self.subtitleSymbolLabel = [[UILabel alloc] initWithFrame:CGRectZero];
         self.subtitleSymbolLabel.userInteractionEnabled = NO;
         [self addSubview:self.subtitleSymbolLabel];
+
+        self.iconImageView = [[UIImageView alloc] initWithFrame:CGRectZero];
+        self.iconImageView.userInteractionEnabled = NO;
+        [self addSubview:self.iconImageView];
     }
     return self;
 }
+
+- (NSString *)subtitle
```

Comment:
Maybe shift to observing these properties on the UILabel and calling -layoutSubviews in one place, rather than conflating model and view a bit here by exposing NSString and UIFont properties which are just passthroughs to the UILabel objects.